feat: util

GeopJr opened the pull request:
A quick util tool to merge and validate translations.

Ideally the validate command should run on PRs and the merge one when there's a change to the default lang.

Validate just checks if the amount of params in the translation is the same as the one in the default on (e.g. `test {} {}` vs `test {}`).

Merge adds/removes strings on all translations based on the default one.

Example runs:

- `node util.mjs validate`

```js
{
  changed: [],
  errors: [
    'zh_CN.json[REMOTE_CRASH_INVALID_TRANSITION_FROM]: incorrect number of parameters (1:2)',
    'zh_CN.json[REMOTE_CRASH_INVALID_BLACKLIST_FROM]: incorrect number of parameters (1:2)'
  ]
}
```

- `node util.mjs merge`

```js
{
  changed: [ 'en_LOL.json', 'zh_CN.json', 'zh_TW.json' ],
  errors: [
    'zh_CN.json[REMOTE_CRASH_INVALID_TRANSITION_FROM]: incorrect number of parameters (1:2)',
    'zh_CN.json[REMOTE_CRASH_INVALID_BLACKLIST_FROM]: incorrect number of parameters (1:2)'
  ]
}
```

Yimura commented:
Translations don't need to be merged as the fallback of en_US.json is used, unless that was intentional to help people to be aware of which keys still need to be translated.

GeopJr commented:
It's intentional, I don't think having translators keep track of en_US to add and remove strings would be ideal:
- typos in keys
- unused keys
- wrong format (`[]` instead of `{}`)

GeopJr commented:
I think it's ready from my side.

- Forced pushed to remove the translation sync (it was only done because this branch's en_US was older than master's)
- Cleaned up the message body
- Added validate check on `%`
- Limited git push to the PR

This is how it looks when a sync happens:

![image](https://user-images.githubusercontent.com/18014039/224371548-293a41bd-45f7-4a49-8203-056313978a38.png)

And this is how it looks when a new language is added (notice that the warnings are about the new language only):

![image](https://user-images.githubusercontent.com/18014039/224371995-c7e60fb0-3895-467c-b280-e99eddbb32a1.png)
